lib/sneakers/tasks.rb

```diff
@@ -41,4 +41,47 @@
 
     r.run
   end
+
+  desc "Retries all failed message for WORKER (Only if using Maxretry Handler)"
+  task :retry_failed_messages do
+
+    if ENV["WORKER"].nil?
+      puts <<EOF
+Error: No worker defined.
+Please set the worker class you want to retry like so:
+
+  $ export WORKER=MyWorker
+  $ rake sneakers:enqueue_failed_messages
+
+or:
+
+  $ rake sneakers:enqueue_failed_messages WORKER=MyWorker
+
+EOF
+    exit(1)
+    end
+
+    worker_class = ENV['WORKER'].constantize
+    config = Sneakers::CONFIG.merge worker_class.queue_opts
+    bunny = Bunny.new(config[:amqp], :vhost => config[:vhost], :heartbeat => config[:heartbeat], :logger => Sneakers::logger)
+    bunny.start
+    channel = bunny.create_channel
+    channel.prefetch config[:prefetch]
+    exchange = channel.exchange(config[:exchange], config[:exchange_options])
+    queue = channel.queue(worker_class.queue_name + "-error", :durable => true)
+
+    delivery_info, properties, payload = queue.pop
```

Comment:
Popping the message off the queue also acks it. So if this rake task fails between this line and when it's published below, this message is lost.

Reply:
You can use manual acknowledgements with `basic.get` (`Bunny::Queue#pop`).